src/main/java/com/zendesk/maxwell/schema/PositionStoreThread.java

```diff
@@ -69,22 +73,24 @@ public void heartbeat() throws Exception {
 		store.heartbeat();
 	}
 
-	private boolean shouldHeartbeat(BinlogPosition heartbeatPosition, BinlogPosition currentPosition) {
+	private boolean shouldHeartbeat(Position heartbeatPosition, Position currentPosition) {
 		if ( currentPosition == null )
 			return true;
 		if ( heartbeatPosition == null )
 			return true;
-		if ( !heartbeatPosition.getFile().equals(currentPosition.getFile()) )
+
+		BinlogPosition heartbeatBinlog = heartbeatPosition.getBinlogPosition();
+		BinlogPosition currentBinlog = currentPosition.getBinlogPosition();
+		if ( !heartbeatBinlog.getFile().equals(currentBinlog.getFile()) )
 			return true;
-		if ( currentPosition.getOffset() - heartbeatPosition.getOffset() > 1000 ) {
+		if ( currentBinlog.getOffset() - heartbeatBinlog.getOffset() > 1000 ) {
```

Comment:
That should probably happen too (and remove RDS heartbeat support at the same time), but this PR is already big enough.